src/context/AppContext.js

```diff
@@ -205,11 +205,11 @@ export const AppProvider = (props) => {
   function checkSession() {
     const localToken = JSON.parse(localStorage.getItem('token'));
     const localUser = JSON.parse(localStorage.getItem('user'));
-    if (localToken && localUser) {
+
+    if (localToken && localUser && session.token) {
```

Comment:
Unfortunately, this doesn't fix the underlying problem, and has the undesirable side effect of logging the user out every time they refresh the page.

We _should_ call `/auth/check_session` if there's a local token and local user, but should ideally suspend other API calls until this check has finished and the user has been logged out (if the response is 401).